### PR TITLE
fix broken build

### PR DESCRIPTION
**Pull request by mganahl**

### CODE REVIEW

File: tensornetwork/block_sparse/blocksparsetensor.py [hunk 1]
```diff
@@ -129,7 +129,7 @@ def shape(self) -> Tuple:
       Tuple: A tuple of `int`.
     """
     return tuple(
-        [reduce(mul, [self._charges[n].dim for n in s]) for s in self._order])
+        reduce(mul, [self._charges[n].dim for n in s]) for s in self._order)
```
Comment (jacksonwb) [inline tensornetwork/block_sparse/blocksparsetensor.py:132]: Maybe update the type hint to Tuple[int]

File: tensornetwork/block_sparse/blocksparsetensor.py [hunk 2]
```diff
@@ -166,11 +166,11 @@ def flows(self) -> List[List]:
 
   @property
   def flat_charges(self) -> List[BaseCharge]:
-    return list([self._charges[o] for o in self.flat_order])
+    return list(self._charges[o] for o in self.flat_order)
 
   @property
   def flat_flows(self) -> List:
-    return list([self._flows[o] for o in self.flat_order])
+    return list(self._flows[o] for o in self.flat_order)
```
Comment (jacksonwb) [inline tensornetwork/block_sparse/blocksparsetensor.py:173]: improve type hint

File: tensornetwork/tests/network_components_free_test.py [hunk 3]
```diff
@@ -6,15 +7,16 @@
...
                                               NodeCollection, AbstractNode,
                                               _remove_trace_edge, _remove_edges)
 import tensornetwork as tn
 from tensornetwork.backends.abstract_backend import AbstractBackend
 from typing import Dict
 
-string_type = h5py.special_dtype(vlen=str)
-
+string_type = network_components.string_type
+ENCODING = network_components.STRING_ENCODING
```
Comment (jacksonwb) [inline tensornetwork/tests/network_components_free_test.py:19]: Maybe use STRING_ENCODING here also for consistency?

File: tensornetwork/tn_keras/test_layer.py [hunk 4]
```diff
@@ -113,7 +114,7 @@ def test_train(dummy_data, make_model):
                 metrics=['accuracy'])
 
   # Train the model for 10 epochs
-  history = model.fit(data, labels, epochs=10, batch_size=32)
+  history = model.fit(data, labels, epochs=50, batch_size=64)
```
Comment (jacksonwb) [inline tensornetwork/tn_keras/test_layer.py:117]: Do these tests not pass at 10 epochs? 50 seems unnecessary.

Comment (mganahl) [reply tensornetwork/tn_keras/test_layer.py:117]: Only some do. After 50 all tests have consistently decreasing loss

Comment (mganahl) [reply tensornetwork/tn_keras/test_layer.py:117]: The problematic tests were only passing by chance before. If you look at the loss you'll see that it is still fluctuating wildly at 10 epochs.  If just happened so that right at 10 epochs the loss was smaller than the initial value, but e.g. at 11 or maybe 12 it was again larger. The new tf release just exposed that.

**Review by jacksonwb — APPROVED** [review]
